### PR TITLE
describe kubelet credentials for authn/authz

### PR DESCRIPTION
**Pull request by deads2k**
Replaces https://github.com/openshift/enhancements/pull/196, which we actually did and forgot to merge.

### CODE REVIEW

File: enhancements/kubelet/kubelet-credentials.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+### How Kubelets Create Credentials
+
+1. If a current kubelet client certificate exists and is valid, use it and skip the rest.
+2. Use the bootstrap kubelet credentials to create a client CertificateSigningRequest (CSR) on the kube-apiserver.
+   The username is in the form `system:node:<nodeName>` and groups contain `system:nodes`
+3. machine-approver (or cluster-admin) approves the CSR created by the kubelet's bootstrap credentials.
+4. Kubelet uses bootstrap kubelet credentials to retrieve the signed certificate from the client CSR.
+5. Given the signed client certificate and the local key (which never left the host), the kubelet communicate with the kube-apiserver
+
+### How machine-approver Decides to Approve Client Certificates
```
Comment (deads2k) [inline enhancements/kubelet/kubelet-credentials.md:128]: @michaelgugino can you confirm this description is correct?

File: enhancements/kubelet/kubelet-credentials.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+1. If the CSR isn't a well-formed kubelet client certificate, don't approve.
+2. If the CSR is for a renewal, don't approve.
```
Comment (michaelgugino) [inline enhancements/kubelet/kubelet-credentials.md:131]: We don't have any logic for 'renewals' specifically for client certs.  Step 4 more or less covers this.

File: enhancements/kubelet/kubelet-credentials.md (hunk 1 above)
Comment (michaelgugino) [reply enhancements/kubelet/kubelet-credentials.md:128]: In addition to this list, we ensure the certificate creation time is not greater than machine-object creation time (TODO: change this to be tied to VM creation time in the future).

**Pull request merged**